core/domain/user_services.py

```diff
@@ -1238,6 +1449,10 @@ def update_user_role(user_id, role):
     if role not in role_services.PARENT_ROLES:
         raise Exception('Role %s does not exist.' % role)
     user_settings = get_user_settings(user_id, strict=True)
+    if user_settings.role == feconf.ROLE_ID_LEARNER:
+        raise Exception('Role update of a Learner is not allowed.')
+    if role == feconf.ROLE_ID_LEARNER:
+        raise Exception('Updating to a Learner is not allowed.')
```

Comment:
Updating to a Learner role is not allowed.

Reply:
Done.